Fix bug of visual feedback when searching gateways

lcc opened the pull request:
Fix bug that caused the app to not show/show the feedback when needed.

pauloserrafh commented on app/src/main/java/br/org/cesar/knot_setup_app/view/gatewayConnected/GatewayConnectedFragment.java:
```diff
@@ -96,6 +96,7 @@ public void run() {
     @Override
     public void onResume() {
         super.onResume();
+        setSearchingFeedback(View.VISIBLE);
```
Rather than doing it here, move to the presenter.
Have `mPresenter.onFocus()` call `feedbackOfGatewayListState()`.

pauloserrafh reviewed (changes requested):
Please rewrite the commit message.
Also, IMO this could be broken in two commits:
1) Fix logic in feedbackOfGatewayListState
2) Add missing visual feedbacks  of gateway search state

Notice that your PR is the whole `Fix bug of visual feedback when searching gateways`, but it might be composed of multiple commits to achieve that (and none of them necessarily must have the same name as the PR).

lcc committed:
Fix logic in feedbackOfGatewayListState

Fix feedbackOfGatewatListState logic to turn visibility on when the
service list is empty and off when the service list isn't empty.

The current version turned the visibility on when the list wasn't empty
which led to a visual feedback of "Looking for gateways..." when there
were already some items in the gateway list.

lcc committed:
Fix visual feedback when searching for gateways

Add checks to the service list current state to provide visual feedback
in key moments (e.g.: when the activity gains focus, when a service has
been resolved).

This is done to give the user an accurate feedback of the state of the
service list.
